python.md

```diff
@@ -0,0 +1,1376 @@
+```python
+unresolved_dependencies = ts.check()
+```
+
+It returns `None` if all dependencies are resolved, or a complex tuple for each
+unresolved dependency. In general, if the check method returns anything but
+`None`, you cannot perform the transaction.
+
+On a dependency failure, check returns a complex tuple of the dependency
+information in the following format:
+
+```
+((N,V,R), (reqN, reqV), needsFlags, suggestedPackage, sense)
+```
+
+The first element is a tuple of the name, version, and release of the package
+you are trying to install. The next tuple holds the required name and required
+version or conflicting name and version. The version will be `None` if the
+dependency is a shared library or other file.
+
+The `needsFlags` tells you about the requirement or conflict. The value is a
+bitmask that can contain the following bit settings: `rpm.RPMSENSE_EQUAL`,
+`rpm.RPMSENSE_GREATER`, and `rpm.RPMSENSE_LESS`. This tells you if the
+dependency is for a version of a package greater than 4.1, for example.
+
+The `suggestedPackage` names a package that solves the dependency. The
+considered packages are those for which you call `addInstall` with a flag of
+`'a'`. This value will be `None` if there is no known package to solve this
+dependency.
+
+You can tell whether the dependency is a conflict or a requirement based on the
+sense value, one of `rpm.RPMSENSE_CONFLICTS` or `rpm.RPMSENSE_REQUIRES`.
+
+For example, the following tuple shows a required package:
+
+```
+(('eruby-devel', '0.9.8', '2'), ('eruby-libs', '0.9.8'), 8, None, 0)
+```
+
+The following tuple shows a required shared library:
+
+```
+(('jpilot', '0.97', '1'), ('libpisock.so.3', None), 0, None, 0)
+```
+
+Note: This tuple format will likely change in future versions of RPM. This
+example shows the format in RPM 4.1. With each RPM release, check the online
+documentation on the Python API to look for changes.
+
+
+## Transaction Check Method Callbacks
+
+You can pass an optional callback function to the call to check. This callback
+gets called for each unresolved dependency in the transaction set. You can use
+this callback to try to automatically bring in required packages, for example.
+
+The basic syntax for the transaction check callback is:
+
+```python
+def check_callback(ts, TagN, N, EVR, Flags):
+    # Do something...
+    pass
+```
+
+Such a callback can be used like this:
+
+```python
+ts = rpm.TransactionSet()
+...
+unresolved_dependencies = ts.check(check_callback)
+```
+
+A full example:
+
+```python
+def check_callback(ts, TagN, N, EVR, Flags):
+    if TagN == rpm.RPMTAG_REQUIRENAME:
+        prev = ""
+    Nh = None
+
+    if N[0] == "/":
+        dbitag = "basenames"
+    else:
+        dbitag = "providename"
+
+    # What do you need to do.
+    if EVR:
+        print("Must find package [", N, "-", EVR, "]")
+    else:
+        print("Must find file [", N, "]")
+
+    # FIXME
+    # if resolved:
+    #     # ts.addIntall(h, h, 'i')
+    #     return -1
```

Comment:
The original example used this undefined `resolved` variable.  I have no clue how it should be set so that the exapmle makes sense.

Reply:
Just delete the example. I don't think anythink has used this interface in 15 years or longer, whether it functions at all is an unknown.